src/vs/platform/environment/common/environment.ts

```diff
@@ -42,6 +42,7 @@ export interface IEnvironmentService {
 
 	execPath: string;
 	appRoot: string;
+	vscodeWindowId: string;
```

Comment:
@Tyriar we cannot really add this here because environment service is agnostic to windows, we have one in the shared process as well as the CLI. I suggest to put this into the `IWindowService` on the renderer side.